lib/srv/db/dynamodb/engine.go

```diff
@@ -478,3 +490,30 @@ func getTargetHeader(req *http.Request) (string, error) {
 	}
 	return target, nil
 }
+
+// NewEndpointsResolver resolves endpoints from DB URI.
+func NewEndpointsResolver(_ context.Context, db types.Database, _ endpoints.ResolverBuilderConfig) (endpoints.Resolver, error) {
+	aws := db.GetAWS()
+	fips := dynamodbutils.IsFIPSEnabled()
+	resolverFns := []resolverFn{
+		resolveDaxEndpoint,
+		resolveDynamoDBEndpoint,
+		resolveDynamoDBStreamsEndpoint,
+	}
```

Comment:
my concern with this, is in cases the agent can only access a subset of these through vpc endpoints.

so the workaround in that case is to manually specify an URI? but when doing that, why do we resolve endpoint first then use the overwrite (same question for the actual access flow)?

Reply:
sounds good 👍